pkg/controllers/edgeconnect/controller.go

```diff
@@ -233,53 +277,63 @@ func (controller *Controller) updateVersionInfo(ctx context.Context, edgeConnect
 		registry.WithKeyChainSecret(&keyChainSecret),
 	)
 	if err != nil {
-		return err
+		_log.Debug("updating finalizers failed", "secretName", keyChainSecret.Name)
+
+		return errors.WithStack(err)
```

Comment:
some errors returned by `registryClientBuilder` are already annotated with stack and some not. So sometimes the stack part will be "doubled"